### PR TITLE
Add: prefer-no-devDependencies

### PR DESCRIPTION
**Pull request by simison**
**Description of change**
- Adds `prefer-no-devDependencies` rule
- Create a new "Disallowed node" category and move `prefer-no-engineStrict` rule there

Happy to add also `prefer-no-dependencies`, `prefer-no-optionalDependencies` and `prefer-no-peerDependencies` if you see this is useful.

Resolves https://github.com/tclindner/npm-package-json-lint/issues/132 (see the issue for use case)

**Checklist**

  - [x] Unit tests have been added
  - [x] Specific notes for documentation, if applicable

### CODE REVIEW

**Comment by simison** [comment]
BTW alternative API for this could be `prefer-no` rule with a list of not allowed rules.

File: docs/rules.md [hunk 1]
```diff
@@ -197,8 +197,9 @@ Add the following to your `.npmpackagejsonlintrc.json` file.
 * [prefer-property-order](rules/package-json-properties/prefer-property-order.md)
 
 
-## Deprecated node rules
+## Disallowed node rules
```
Comment (tclindner) [inline docs/rules.md]: 👍

File: docs/rules/disallowed-nodes/prefer-no-devDependencies.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (tclindner) [inline docs/rules/disallowed-nodes/prefer-no-devDependencies.md]: Thank you so much for adding docs 🎉

**Comment by tclindner** [comment]
I would love to add `prefer-no-dependencies`, `prefer-no-optionalDependencies`, and` prefer-no-peerDependencies` if you have time. Thank you so much for offering 🥇

**Pull request merged**